### ISSUE TITLE
warning: already initialized constant Digest::REQUIRE_MUTEX

### ISSUE THREAD

**Issue by remy727** [ruby/digest#44]
I know this is already fixed but I am getting this issue with digest `3.1.1`

```
/Users/admin/.rvm/gems/ruby-2.5.1/gems/digest-3.1.1/lib/digest.rb:20: warning: already initialized constant Digest::REQUIRE_MUTEX
/Users/admin/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/digest.rb:6: warning: previous definition of REQUIRE_MUTEX was here
```

**Comment by mzanini** [comment]
Same issue, could you point me to where this was fixed @remy727? I have just installed the latest version and this warning pops up every time I start my rails server.

**Comment by remy727** [comment]
@mzanini, I fixed by upgrading ruby version to `3.2.0`

**Comment by and-enaruta** [comment]
@mzanini The PR was closed https://github.com/ruby/digest/pull/36.

**Comment by and-enaruta** [comment]
@remy727 Is there a way to fix it for early ruby versions? Like 2.7 at least.

**Comment by macanudo527** [comment]
@and-enaruta You could hack the gem with [this fix](https://stackoverflow.com/questions/7953611/solutions-to-the-annoying-warning-already-initialized-constant-message) to get rid of the error. I haven't tested it myself though, since it should be temporary for me. I'm working on upgrading to Ruby 3.2+
